### DISCUSSION TITLE
A way to force musl platform download if available?

### DISCUSSION THREAD

**Discussion by KarateSnowMachine** [mason-org/mason.nvim#843] [Q&A]
Hello, 

I am using Mason on RHEL7 and I ran into an issue where Mason installs the gnu/glibc version of rust-analyzer (which is a perfectly good thing to do). However, when running the binary I get a glibc version error since RHEL7 is ancient. My workaround was simply to hack the package file to just change gnu platform to download the musl package: 

```
 when(platform.is.linux_x64_gnu, "rust-analyzer-x86_64-unknown-linux-musl.gz")
```
And this works fantastically well for me. 

However, I wasn't sure if there was simply a way to override the platform detection using a config option so I don't have to keep this local code hack to Mason. 

Thanks for your help and a wonderful project!

**Comment by williamboman** [comment]
Hello! No not yet. This will however be possible after https://github.com/mason-org/mason-registry is integrated. It'll work like so:

```vim
:MasonInstall --target=linux_x64 rust-analyzer
```

or alternatively

```lua
require("mason-registry").get_package("rust-analyzer"):install { target = "linux_x64" }
```
